spec/models/admin_dashboard_data_spec.rb

```diff
@@ -3,37 +3,129 @@
 require 'rails_helper'
 
 describe AdminDashboardData do
+  after do
+    AdminDashboardData.reset_problem_checks
+    Discourse.redis.flushdb
+  end
+
+  describe "#fetch_problems" do
+    describe "adding problem messages" do
+      it "adds the message and returns it when the problems are fetched" do
+        AdminDashboardData.add_problem_message("dashboard.bad_favicon_url")
+        problems = AdminDashboardData.fetch_problems.map(&:to_s)
+        expect(problems).to include(I18n.t("dashboard.bad_favicon_url", { base_path: Discourse.base_path }))
+      end
+
+      it "does not allow adding of arbitrary problem messages, they must exist in @problem_messages" do
+        AdminDashboardData.add_problem_message("errors.messages.invalid")
+        problems = AdminDashboardData.fetch_problems.map(&:to_s)
+        expect(problems).not_to include(I18n.t("errors.messages.invalid"))
+      end
+    end
+
+    describe "adding new checks" do
+      it 'calls the passed block' do
+        called = false
+        AdminDashboardData.add_problem_check do
+          called = true
+        end
+
+        AdminDashboardData.fetch_problems
+        expect(called).to eq(true)
+
+        AdminDashboardData.fetch_problems(check_force_https: true)
+        expect(called).to eq(true)
```

Comment:
I think it is unclear what we're testing for here. `called` is `true` after the initial call to `AdminDashboardData.fetch_problems` but I'm not sure if it should remain `true` after we call `AdminDashboardData.fetch_problems(check_force_https: true)`? Do we have to reset `called` to `false` first?

Reply:
Don't know what the second call with `check_force_https` was trying to achieve so I removed it